flake.nix

```diff
@@ -28,6 +30,239 @@
     };
 
     pkgs = import nixpkgs { system = "x86_64-linux"; overlays = builtins.attrValues overlays; };
+    darwin-pkgs = import nixpkgs { system = "x86_64-darwin"; overlays = builtins.attrValues overlays; };
+
+
+    # we can use this, to get a coherent picture of the sources for
+    # the various libraries.  The following command will produce a
+    # JSON output, that contains each of our libs, with their respective
+    # versions.
+    #
+    #    nix eval --json .#lib-srcs
+    #
+    lib-srcs = {
+      secp256k1 = pkgs.secp256k1.src.url;
+      sodium    = pkgs.libsodium-vrf.src.url;
+      blst      = pkgs.libblst.src.url;
+    };
+
+    dist = let
+      # For packaging, we can'd deal with split outputs.
+      mkSingleOutput = drv: drv.overrideDerivation (drv': { outputs = [ "out" ]; });
```

Comment:
How does this work?  If a derivation's `buildPhase` or something expects there to be multiple outputs like `$out`, `$doc` and `$dev` is it not likely to still attempt to use those undefined env vars with unpredictable results?

Reply:
This just overrides the output for the given drv. And yes it only works with the auto-split stuff if someone explicitly creates the $xxx, this will fail.